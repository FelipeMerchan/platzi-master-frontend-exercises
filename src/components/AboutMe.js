import React from 'react'

import './styles/AboutMe.css'

function AboutMe() {
  return (
    <article className="AboutMe">
      <h2 className="AboutMe__title">quien
        <br /> <span>soy</span>
      </h2>
      <p className="AboutMe__description">
        Etiam lectus nulla, vestibulum vel luctus eu, 
        ultricie Sit amet arcu. In a sem a nibh fringilla 
        blandit. Ut u Metus turpis ultricies.
      </p>
    </article>
  )
}

export default AboutMe