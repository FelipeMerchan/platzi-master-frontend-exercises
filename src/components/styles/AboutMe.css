.AboutMe {
  font-family: var(--font-4);
  color: #07080d;
}

.AboutMe__title {
  color: #c5345c;
}

.AboutMe__title>span {
  color: #66615f;
}