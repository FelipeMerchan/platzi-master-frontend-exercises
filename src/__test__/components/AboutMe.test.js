import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import AboutMe from '../../components/AboutMe'

describe('Probar el componente <AboutMe />', () => {
  const aboutMe = shallow(<AboutMe />)
  test('Render del componente AboutMe', () => {
    expect(aboutMe.length).toEqual(1)
  })
  test('Render con el className AboutMe__title', () => {
    expect(aboutMe.exists('.AboutMe__title')).toBeTruthy()
  })
  test('Render de la descripciÃ³n del autor', () => {
    expect(aboutMe.find('.AboutMe__description').text()).toEqual('Etiam lectus nulla, vestibulum vel luctus eu, ultricie Sit amet arcu. In a sem a nibh fringilla blandit. Ut u Metus turpis ultricies.')
  })
})

describe('AboutMe Snapshot', () => {
  const aboutMe = create(<AboutMe />)
  test('Comprobar la UI del componente AboutMe', () => {
    expect(aboutMe.toJSON()).toMatchSnapshot()
  })
})