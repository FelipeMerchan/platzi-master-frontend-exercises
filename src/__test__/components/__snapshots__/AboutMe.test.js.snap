// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMe Snapshot Comprobar la UI del componente AboutMe 1`] = `
<article
  className="AboutMe"
>
  <h2
    className="AboutMe__title"
  >
    quien
    <br />
     
    <span>
      soy
    </span>
  </h2>
  <p
    className="AboutMe__description"
  >
    Etiam lectus nulla, vestibulum vel luctus eu, ultricie Sit amet arcu. In a sem a nibh fringilla blandit. Ut u Metus turpis ultricies.
  </p>
  <a
    className="AboutMe__link bg-gray-gradient"
    href="#readMore"
    id="readMore"
  >
    leer m√°s
  </a>
</article>
`;
